odoo/addons/base/models/ir_actions.py

```diff
@@ -1170,57 +1170,63 @@ def run(self):
         """
         res = False
         for action in self.sudo():
-            action_groups = action.group_ids
-            if action_groups:
-                if not (action_groups & self.env.user.all_group_ids):
-                    raise AccessError(_("You don't have enough access rights to run this action."))
-            else:
-                model_name = action.model_id.model
-                try:
-                    self.env[model_name].check_access("write")
-                except AccessError:
-                    _logger.warning("Forbidden server action %r executed while the user %s does not have access to %s.",
-                        action.name, self.env.user.login, model_name,
-                    )
-                    raise
-
             eval_context = self._get_eval_context(action)
             records = eval_context.get('record') or eval_context['model']
             records |= eval_context.get('records') or eval_context['model']
-            self._can_execute_action_on_records(records)
-
-            if action.warning:
-                raise ServerActionWithWarningsError(_("Server action %(action_name)s has one or more warnings, address them first.", action_name=action.name))
+            action._can_execute_action_on_records(records)
+            res = action._run(records, eval_context)
+        return res
 
-            runner, multi = action._get_runner()
-            if runner and multi:
-                # call the multi method
-                run_self = action.with_context(eval_context['env'].context)
+    def _run(self, records, eval_context):
```

Comment:
I'm not sure I understand the need for this commit at all, how does this "add granularity" ?

We're splitting some methods here but none of them (run, _run, or _can_execute_action_on_records) are overridden in the enterprise changes.

Would help if there was a commit message.

Reply:
> I'm not sure I understand the need for this commit at all, how does this "add granularity" ?

That's what allow us to skip group check on AI tools

> are overridden in the enterprise changes.

But we call them

I updated the commit message to add more context